algebird-test/src/test/scala/com/twitter/algebird/BytesSpec.scala

```diff
@@ -0,0 +1,89 @@
+package com.twitter.algebird
+
+import org.scalacheck.Gen
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+import org.scalatest.{Matchers, WordSpec}
+
+class BytesSpec extends WordSpec with Matchers with GeneratorDrivenPropertyChecks {
+
+  "has a sane equals function" in {
+    val random = new scala.util.Random
+    forAll((Gen.choose(1, 4096), "wordLength")) { (wordLength: Int) =>
+      // Given two different array instances that have the same content.
+      val word = random.nextString(wordLength)
```

Comment:
Good catch regarding the `equals` test.  In the latest commit I enhanced the test spec of Bytes in general.

> compare(a, b) == 0 would pass your tests

As part of the changes I also enhanced the `compare` tests, although `compare(a, b) == 0` was already failing the tests prior to the enhancements.  Hence I'm not sure but hope that the enhancements will still cover any gap you found yesterday for `compare`.